File: bms/go/geo/geo_s1_Interval_ComplementCenter/geo_s1_Interval_ComplementCenter.go
package test

import (
	"math"
)

func (i geo_s1_Interval) ComplementCenter() float64 {
	if i.Lo != i.Hi {
		return i.Complement().Center()
	}
	// Singleton. The interval just contains a single point.
	if i.Hi <= 0 {
		return i.Hi + math.Pi
	}
	return i.Hi - math.Pi
}

func (i geo_s1_Interval) Complement() geo_s1_Interval {
	if i.Lo == i.Hi {
		// Singleton. The interval just contains a single point.
		return geo_s1_FullInterval()
	}
	// Handles empty and full.
	return geo_s1_Interval{i.Hi, i.Lo}
}

func geo_s1_FullInterval() geo_s1_Interval { return geo_s1_Interval{-math.Pi, math.Pi} }

func (i geo_s1_Interval) Center() float64 {
	c := 0.5 * (i.Lo + i.Hi)
	if !i.IsInverted() {
		return c
	}
	if c <= 0 {
		return c + math.Pi
	}
	return c - math.Pi
}

func (i geo_s1_Interval) IsInverted() bool { return i.Lo > i.Hi }

type geo_s1_Interval struct {
	Lo, Hi float64
}

File: translations_only/fail/go/geo/geo_s1_Interval_ComplementCenter/claude3/BaseRepair/geo_s1_Interval_ComplementCenter.rs

use std::f64::consts::PI;

struct GeoS1Interval {
    lo: f64,
    hi: f64,
}

fn complement_center(i: GeoS1Interval) -> f64 {
    if i.lo != i.hi {
        return center(complement(i));
    }
    // Singleton. The interval just contains a single point.
    if i.hi <= 0.0 {
        return i.hi + PI;
    }
    return i.hi - PI;
}

fn complement(i: GeoS1Interval) -> GeoS1Interval {
    if i.lo == i.hi {
        // Singleton. The interval just contains a single point.
        return geo_s1_full_interval();
    }
    // Handles empty and full.
    return GeoS1Interval { lo: i.hi, hi: i.lo };
}

fn geo_s1_full_interval() -> GeoS1Interval {
    return GeoS1Interval { lo: -PI, hi: PI };
}

fn center(i: GeoS1Interval) -> f64 {
    let c = 0.5 * (i.lo + i.hi);
    if !is_inverted(i) {
        return c;
    }
    if c <= 0.0 {
        return c + PI;
    }
    return c - PI;
}

fn is_inverted(i: GeoS1Interval) -> bool {
    return i.lo > i.hi;
}
